### PR TITLE
Fix func update for brew and add uninstall

### LINKED ISSUE #369: Add command to uninstall azure functions core tools
Since we prompt the user to install the Azure Functions Core Tools, it feels like we should also provide an 'uninstall' command. I figure something in the command palette would be fine.

I also just want this for myself because I have to uninstall every time I want to test install 😂

### LINKED ISSUE #388: prompt to upgrade to 2.0.1-beta.26 doesn't install 26
When the extension activates I get this message:

image

I click `Update` and then it runs `brew tap azure/function`, but it seems like it is trying to install .25.  This is the 2nd time I tried it, so there is less output. The first time did a little more work:

```
Running command: "brew tap azure/functions"...
Finished running command: "brew tap azure/functions".
Running command: "brew install azure-functions-core-tools"...
Warning: azure/functions/azure-functions-core-tools 2.0.1-beta.25 is already installed and up-to-date
To reinstall 2.0.1-beta.25, run `brew reinstall azure-functions-core-tools`
Finished running command: "brew install azure-functions-core-tools".
```

### PR DESCRIPTION
**Pull request by EricJizbaMSFT**
I did a little refactoring to split the 'funcCoreTools' code up, but the main changes are in these two files:

1. updateFuncCoreTools.ts: Use 'brew update' instead of 'brew install' when updating 
Fixes #388

1. uninstallFuncCoreTools.ts: Add the ability for uninstalling the tools. This is mostly for our sake when testing, but a few users might find this in the command palette and use it as well.
Fixes #369

### CODE REVIEW

**Comment by nturinski** [comment]
Should we allow `Install Function Core Tools` to be exposed on the command palette if uninstall is?

File: src/funcCoreTools/tryGetLocalRuntimeVersion.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import { isWindows, ProjectRuntime } from '../constants';
+import { getLocalFuncCoreToolsVersion } from './getLocalFuncCoreToolsVersion';
+
+export async function tryGetLocalRuntimeVersion(): Promise<ProjectRuntime | undefined> {
+    if (!isWindows) {
+        return ProjectRuntime.beta;
+    } else {
+        try {
+            const version: string | null = await getLocalFuncCoreToolsVersion();
+            if (version !== null) {
```
[the remark this answers is not in the document]
Comment (EricJizbaMSFT) [reply src/funcCoreTools/tryGetLocalRuntimeVersion.ts]: "not my code" but I'll fix it :) Just copied from one file to a separate file.

I agree in this case that `!version` is better. However, I think there are pretty common cases where explicitly checking against null or undefined makes sense (specifically variables of type `enum | undefined` or `number | undefined`). Plus we have strictNullChecks on, so I'm not too worried about mixing up null and undefined

File: src/utils/functionRuntimeUtils.ts [hunk 2]
```diff
@@ -9,133 +9,88 @@ import request = require('request-promise');
...
-                    return;
-                }
-                this.properties.localVersion = localVersion;
-                const major: number = semver.major(localVersion);
-                const newestVersion: string | undefined = await getNewestFunctionRuntimeVersion(major, this);
-                if (!newestVersion) {
-                    return;
-                }
+            const projectRuntime: ProjectRuntime | undefined = getProjectRuntimeFromVersion(localVersion);
+            if (projectRuntime === undefined) {
```
[the remark this answers is not in the document]
Comment (EricJizbaMSFT) [reply src/utils/functionRuntimeUtils.ts]: This is an example where it's an enum and I prefer `=== undefined`

**Comment by EricJizbaMSFT** [comment]
@nturinski I'd consider that another feature. Feel free to create an issue if you think it's worth doing

**Pull request merged**